pkg/comp-functions/functions/vshnredis/release.go

```diff
@@ -0,0 +1,199 @@
+// from other jobs, such as maintenance, are reflected in the release.
+func getRelease(ctx context.Context, iof *runtime.Runtime) (*v1beta1.Release, error) {
+	r := &v1beta1.Release{}
+	err := iof.Observed.Get(ctx, r, redisRelease)
+	if err != nil && err != runtime.ErrNotFound {
+		return nil, fmt.Errorf("cannot get redis release from observed iof: %v", err)
+	}
+
+	if err == runtime.ErrNotFound {
+		err := iof.Desired.Get(ctx, r, redisRelease)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get redis release from desired iof: %v", err)
+		}
+	}
+	return r, nil
+}
+
+func updateRelease(ctx context.Context, comp *vshnv1.VSHNRedis, releaseName string, values map[string]interface{}) error {
+	l := controllerruntime.LoggerFrom(ctx)
+
+	l.V(1).Info("Setting release version")
+	err := setReleaseVersion(comp, values)
+	if err != nil {
+		return fmt.Errorf("cannot set redis version for release %s: %v", releaseName, err)
+	}
+
+	l.V(1).Info("Adding PVC annotations")
+	err = addPVCAnnotation(values)
+	if err != nil {
+		return fmt.Errorf("cannot add pvc annotations for release %s: %v", releaseName, err)
+	}
+
+	l.V(1).Info("Adding pod annotations")
+	err = addPodAnnotation(values)
+	if err != nil {
+		return fmt.Errorf("cannot add pod annotations for release %s: %v", releaseName, err)
+	}
+
+	l.V(1).Info("Adding backup config map")
+	err = addBackupCM(values)
+	if err != nil {
+		return fmt.Errorf("cannot add configmap for release %s: %v", releaseName, err)
+	}
+	return nil
+}
+
+func addPVCAnnotation(valueMap map[string]any) error {
+	annotations := map[string]interface{}{
+		"k8up.io/backup": "false",
+	}
+	err := unstructured.SetNestedMap(valueMap, annotations, "master", "persistence", "annotations")
+	if err != nil {
+		return fmt.Errorf("cannot set annotations the helm values for key: master.persistence")
+	}
+
+	return nil
+}
+
+func addPodAnnotation(valueMap map[string]any) error {
+	annotations := map[string]interface{}{
+		"k8up.io/backupcommand":  "/scripts/backup.sh",
+		"k8up.io/file-extension": ".tar",
+	}
+	err := unstructured.SetNestedMap(valueMap, annotations, "master", "podAnnotations")
+	if err != nil {
+		return fmt.Errorf("cannot set annotations the helm values for key: master.podAnnotations")
+	}
+
+	return nil
+}
+
+func addBackupCM(valueMap map[string]any) error {
+	masterMap, ok := valueMap["master"].(map[string]any)
+	if !ok {
+		return fmt.Errorf("cannot parse the helm values for key: master")
+	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: backupScriptCMName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: backupScriptCMName,
+					},
+					DefaultMode: pointer.Int32(0774),
+				},
+			},
+		},
+	}
+
+	masterMap["extraVolumes"] = volumes
+
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      backupScriptCMName,
+			MountPath: "/scripts",
```

Comment:
is it some general VSHN script or is it under our control?

Reply:
This should be under out control. @Kidswiss should have more to say about that.

Reply:
It's our script, see: https://github.com/vshn/appcat/blob/master/pkg/comp-functions/functions/vshnredis/script/backup.sh